roles/sushy_emulator/tasks/collect_details.yml

```diff
@@ -53,24 +61,41 @@
     dest: "{{ cifmw_sushy_emulator_resource_directory }}/known_hosts"
     mode: '0644'
 
-- name: Get details for Libvirt driver
-  when: cifmw_sushy_emulator_driver == 'libvirt'
+- name: Load cifmw_libvirt_manager_uuids fact from file
+  vars:
+    _uuid_file: "{{ cifmw_sushy_emulator_basedir }}/artifacts/libvirt-uuids.yml"
+  when:
+    - cifmw_libvirt_manager_uuids is not defined
   block:
-    - name: Set vars
-      ansible.builtin.set_fact:
-        _libvirt_uri: "qemu+ssh://{{ cifmw_sushy_emulator_libvirt_user }}@{{ hostvars[cifmw_baremetal_hypervisor_target].ansible_host }}/system"
+    - name: "Slurp content of: {{ _uuid_file }}"
+      ansible.builtin.slurp:
+        src: "{{ _uuid_file }}"
+      register: _libvirt_uuids_file
 
-    # todo(Lewis): Once changes are made to the libvirt_manager role, the baremetal-info.yml can be consumed to gather required info
-    - name: Get Libvirt instance UUIDs
-      ansible.builtin.shell:
-        cmd: |
-          set -o pipefail
-          virsh --connect={{ _libvirt_uri }} list --all --uuid --name | grep {{ cifmw_sushy_emulator_instance_node_name_prefix }} | cut -d' ' -f1
-      register: _virsh_list_uuid
+    - name: "Set cifmw_baremetal_hosts fact from {{ _uuid_file }}"
+      vars:
+        _yaml: "{{ _libvirt_uuids_file.content | b64decode | from_yaml }}"
+      ansible.builtin.set_fact:
+        cifmw_libvirt_manager_uuids: "{{ _yaml.libvirt_uuid }}"
 
-    - name: Set instance_uuid variable
+- name: Set fact related to Libvirt driver
+  when:
+    - cifmw_sushy_emulator_driver == 'libvirt'
+  block:
+    - name: Generate list of filtered VMs
+      vars:
+        _matching_vms: >-
+          {% set matching_vms = [] -%}
+          {% for host, uuid in cifmw_libvirt_manager_uuids.items() -%}
+          {%   if host | regex_search(cifmw_sushy_emulator_vm_prefix_filter | default('') + '.*') -%}
```

Comment:
Can the default not simply be `| default('.*')`

Reply:
The two cases I'm aiming for here are:
- You provide `cifmw_sushy_emulator_vm_prefix_filter` like `cifmw-baremetal` so the regex would be `cifmw-baremetal.*`
- You don't provide a `cifmw_sushy_emulator_vm_prefix_filter` so the regex is just `.*`

If I do `| default('.*')` then the user would need to set `cifmw_sushy_emulator_vm_prefix_filter` to `cifmw-baremetal.*` being slightly less user friendly. 

Please let me know if you would still like it changed.

Reply:
oh my, ok. I did not see that. Let's keep it.

Unless we can use `startswith` ?

`{% if host.startswith(cifmw_sushy_emulator_vm_prefix_filter) %}` ?